From 048f8c3b9b4fbfcad75ac049c5ba24e1ead3a19b Mon Sep 17 00:00:00 2001
From: Victor Westerhuis <victor@westerhu.is>
Date: Sun, 19 Jun 2022 17:33:56 +0200
Subject: [PATCH] Use correct systemd system unit directory

Keep the default the same, so there is no change in behavior.

See: https://www.freedesktop.org/software/systemd/man/systemd.html#System%20unit%20directories
---
 meson.build | 10 ++++++----
 1 file changed, 6 insertions(+), 4 deletions(-)

diff --git a/meson.build b/meson.build
index 4544587..8dfe10d 100644
--- a/meson.build
+++ b/meson.build
@@ -37,7 +37,9 @@ sysconfdir = get_option('sysconfdir') / meson.project_name()
 libexecdir = get_option('libexecdir') / meson.project_name()
 mandir = get_option('mandir')
 moduledir = get_option('libdir') / meson.project_name()
-systemddir = get_option('libdir') / 'systemd'
+
+systemd_deps = dependency('systemd', required: false)
+systemdsystemunitdir = systemd_deps.get_variable('systemdsystemunitdir', default_value: get_option('libdir') / 'systemd/system')
 
 #
 # Required dependencies
@@ -150,7 +152,7 @@ summary({
   'sysconfdir': sysconfdir,
   'moduledir': moduledir,
   'mandir': mandir,
-  'systemddir': systemddir,
+  'systemdsystemunitdir': systemdsystemunitdir,
 }, section: 'Directories')
 summary({
   'extra_debug': get_option('extra_debug'),
@@ -183,11 +185,11 @@ foreach filename, kwargs : {
     'install_mode': 'rwxr-xr-x',
   },
   'docs/kmscon.service.in': {
-    'install_dir': systemddir / 'system',
+    'install_dir': systemdsystemunitdir,
     'install_mode': 'rw-r--r--',
   },
   'docs/kmsconvt@.service.in': {
-    'install_dir': systemddir / 'system',
+    'install_dir': systemdsystemunitdir,
     'install_mode': 'rw-r--r--',
   },
 }
-- 
2.37.2

